tests/run-pass/no_main.rs

```diff
@@ -0,0 +1 @@
+#![no_main]
```

Comment:
you may be able to make this test pass in `compile-fail` by adding a `// error-pattern miri can only run programs that have a main function` comment to this file

Reply:
Ah yes that works :+1: